dxr/plugins/rust/refs.py

```diff
@@ -126,6 +126,29 @@ def menu_items(self):
         return menu
 
 
+class MacroRef(_RustRef):
+    def prepare_menu_data(self, tree_index, datum):
+        self.hover = datum['qualname']
+        return trim_dict(datum, ['qualname'])
+
+    def menu_items(self):
+        return [find_references_menu_item(self.tree, self.menu_data["qualname"], "macro-ref", "macro")]
+
+class MacroRefRef(_RustRef):
+    def prepare_menu_data(self, tree_index, datum):
+        self.hover = datum['qualname']
```

Comment:
Another, slightly less imperative way to narrow down the `datum` dict to just the "qualname" key would be to subclass `_KeysFromDatum` rather than `_RustRef` and say `keys = ['qualname']` out at the class level. The nice thing about that is that it becomes obvious from examining the MacroRefRef class what it stores in elasticsearch. However, most other things still call trim_dict, so it's not a blocker.